smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/rules/ContentPublishRule.java

```diff
@@ -264,38 +270,58 @@ private void doReplicationChecks() throws SmokeTestException {
             polling = new Polling(() -> {
                 agentsRef.set(replicationClient.getAgentQueueJson());
                 log.info("Replication agents list: {}", agentsRef.get());
-
-                return ReplicationClient.checkDistributionAgentExists(agentsRef.get(), PUBLISH_DIST_AGENT);
+                boolean internalPublishAgentExists = ReplicationClient.checkDistributionAgentExists(agentsRef.get(), INTERNAL_PUBLISH_DIST_AGENT);
+                if (!internalPublishAgentExists) {
+                    log.info("Internal publish agent does not exist");
+                    this.publishDistAgent = PUBLISH_DIST_AGENT;
+                    return ReplicationClient.checkDistributionAgentExists(agentsRef.get(), PUBLISH_DIST_AGENT);
+                }
+                return internalPublishAgentExists;
             });
             polling.poll(TIMEOUT, 500);
         } catch (TimeoutException e) {
             throw replicationClient.getReplicationException(REPLICATION_NOT_AVAILABLE,
-                String.format("Replication agent %s unavailable", PUBLISH_DIST_AGENT), polling.getLastException());
+                String.format("Replication agent %s unavailable", this.publishDistAgent), polling.getLastException());
         } catch (InterruptedException | RuntimeException e) {
             throw replicationClient.getGenericException("Replication agent unavailable", e);
         }
 
         Agents agents = agentsRef.get();
         
-        // throw if publish agent is blocked
-        boolean agentQueueBlocked = ReplicationClient.isAgentQueueBlocked(agents, PUBLISH_DIST_AGENT);
+        boolean agentQueueBlocked = ReplicationClient.isAgentQueueBlocked(agents, this.publishDistAgent);
```

Comment:
@tmaret I noticed that sometimes `isAgentQueueBlocked` returns false although before running the smoke tests the queue was blocked. This is why the `PublishEndToEndIT` test sometimes fails with Timeout instead of  BlockedQueue. Do you have any idea what could be wrong? Should I create a separate issue for this?

Reply:
Good catch. My hunch is that this is related to caching. I suggest to file a ticket to investigate the root cause.

Reply:
https://jira.corp.adobe.com/browse/SKYOPS-57117